### ISSUE TITLE
DatePicker: Month switches sides upon date selection in RangePicker

### ISSUE THREAD

**Issue by jimcorban** [NG-ZORRO/ng-zorro-antd#6322]
### Reproduction link
[https://stackblitz.com/edit/itzefl--run?file=src/app/app.component.ts](https://stackblitz.com/edit/itzefl--run?file=src/app/app.component.ts)

### Steps to reproduce
When clicking into the date range picker, a dropdown is displayed that includes the calendars for two months, side-by-side. Click on any date in the right-side month, and note that that month will be immediately moved to the left hand side.

### What is expected?
When selecting a day on one of the month calendars shown in the date range picker, that month should remain where it is.

### What is actually happening?
Click on any date in the right-side month when selecting a start date on an empty date range, and note that that month will be immediately moved to the left hand side.

| Environment | Info |
|---|---|
| ng-zorro-antd | 11.0.1 |
| Browser | Chrome - latest |

---
This applies to the regular date range picker (i.e., Jan 1, 2021 - Jan 7, 2021). I haven't tested it on the time / week / month pickers. 

The React version of this control works as expected - the month calendars remain where they are as you select dates. https://ant.design/components/date-picker/

**Comment by jimcorban** [comment]
https://user-images.githubusercontent.com/28040084/103936261-3321e100-50f5-11eb-8710-f2405ecb216d.mov


https://user-images.githubusercontent.com/28040084/103936268-35843b00-50f5-11eb-9b17-7b3d81517856.mov

Here's an example recording from the component demo page, along with one demonstrating the expected behavior from the React component demo page.

**Comment by jimcorban** [comment]
This has been assigned the wrong label - it's the DatePicker component, not Select.

**Comment by jimcorban** [comment]
The users of my app complain about this a lot - it really makes the RangePicker almost unusable.

**Comment by HDaghash** [comment]
I agree with @jimcorban especially on date edit it's lead to swap the months in the calendar which makes no sense and confuse the user, please check the GIF :


![ezgif com-gif-maker](https://user-images.githubusercontent.com/15608990/111062093-4455e800-84c0-11eb-8989-ed8cbe48014f.gif)

**Comment by zyuelei** [comment]
+1
